editors/code/src/run.ts

```diff
@@ -104,24 +104,34 @@ export function prepareEnv(
 }
 
 export async function createTask(runnable: ra.Runnable, config: Config): Promise<vscode.Task> {
-    if (runnable.kind !== "cargo") {
-        // rust-analyzer supports only one kind, "cargo"
-        // do not use tasks.TASK_TYPE here, these are completely different meanings.
-
-        throw `Unexpected runnable kind: ${runnable.kind}`;
+    let definition: tasks.CargoTaskDefinition;
+    let args: string[];
+    if (runnable.kind === "cargo") {
+        const cargoArgs = runnable.args as ra.CargoRunnable;
+        args = createCargoArgs(cargoArgs);
+
+        definition = {
+            type: tasks.TASK_TYPE,
+            command: args[0], // run, test, etc...
+            args: args.slice(1),
+            cwd: cargoArgs.workspaceRoot || ".",
+            env: prepareEnv(runnable, config.runnablesExtraEnv),
+            overrideCargo: cargoArgs.overrideCargo,
+        };
+    } else {
+        const projectJsonRunnableArgs = runnable.args as ra.ProjectJsonRunnable;
+        args = projectJsonRunnableArgs.args;
+
+        definition = {
+            type: tasks.TASK_TYPE,
+            command: args[0],
+            args: args.slice(1),
+            cwd: projectJsonRunnableArgs.workspaceRoot,
+            env: prepareEnv(runnable, config.runnablesExtraEnv),
+            overrideCargo: "buck2",
```

Comment:
Does this mean only `buck2` is supported for projectjson runnables?

Reply:
Yeah, ideally we'd fetch `buck2` string from `projectJsonRunnableArgs`

Reply:
Oh, yeah, sorry—I meant to pull this string from configuration. Didn’t realize I left it here.